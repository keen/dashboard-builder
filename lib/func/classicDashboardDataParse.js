export default function classicDashboardDataParse(oldDashboard, chartId) {
  const typeForSelect = (item = '') => {
    if (!item) return '';
    const label = item.replace(/-/g, ' ');
    return label.charAt(0).toUpperCase() + label.slice(1);
  };
  let rowTitle;
  let topPosition = 0;
  const firstOldDashboards = {
    ...oldDashboard,
    settings: {
      items: oldDashboard.rows.map(r => {
        topPosition += Math.floor(r.height/100);
        if (r.title) {
          rowTitle = {
            i: chartId,
            type: 'paragraph',
            text: `<h1>${r.title}</h1>`,
            y: topPosition - Math.floor(r.height/100),
            x: 0,
            h: 1,
            w: 12
          };
          topPosition += 2;
        }
        const elements = r.tiles.map((t, j) => {
          let prevElWidth = 0;
          if (j !== 0) {
            prevElWidth = r.tiles[j - 1].column_width;
          }
          let elHeight = Math.floor(r.height/100);
          return {
            i: chartId,
            h: elHeight * 4,
            w: t.column_width,
            y: topPosition - elHeight,
            x: prevElWidth,
            sparkline: false,
            savedQuery: [
              {
                value: t.query_name,
                label: typeForSelect(t.query_name)
              }
            ]
          };
        });
        if (r.title) {
          elements.unshift(rowTitle);
        }
        return elements;
      }),
      dryRun: false
    },
  };
  let oneArray = [];
  firstOldDashboards.settings.items.forEach(r => {
    oneArray.push(...r);
  });
  const translateOldDashboards = {
    ...firstOldDashboards,
    settings: {
      ...firstOldDashboards.settings,
      items: oneArray,
      layout: oneArray,
    }
  };
  return translateOldDashboards;
}
