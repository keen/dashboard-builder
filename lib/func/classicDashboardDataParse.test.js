import classicDashboardDataParse from './classicDashboardDataParse';

describe('classicDashboardDataParse()', () => {
  it('should parse data correctly', () => {
    const dashboardInfo = {
      rows: [
        {
          height: 280,
          tiles: [
            {
              column_width: 6,
              query_name: 'users'
            },
            {
              column_width: 6,
              query_name: 'sales'
            }
          ],
          title: 'Users and sales'
        }
      ]
    }
    const parsedData = classicDashboardDataParse(dashboardInfo, '19c7a585-82ac-4e83-bbd4-b8757b445c4e');
    expect(parsedData).toMatchSnapshot();
  });
});