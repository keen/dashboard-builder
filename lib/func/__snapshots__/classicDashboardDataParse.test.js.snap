// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classicDashboardDataParse() should parse data correctly 1`] = `
Object {
  "rows": Array [
    Object {
      "height": 280,
      "tiles": Array [
        Object {
          "column_width": 6,
          "query_name": "users",
        },
        Object {
          "column_width": 6,
          "query_name": "sales",
        },
      ],
      "title": "Users and sales",
    },
  ],
  "settings": Object {
    "dryRun": false,
    "items": Array [
      Object {
        "h": 1,
        "i": "19c7a585-82ac-4e83-bbd4-b8757b445c4e",
        "text": "<h1>Users and sales</h1>",
        "type": "paragraph",
        "w": 12,
        "x": 0,
        "y": 0,
      },
      Object {
        "h": 8,
        "i": "19c7a585-82ac-4e83-bbd4-b8757b445c4e",
        "savedQuery": Array [
          Object {
            "label": "Users",
            "value": "users",
          },
        ],
        "sparkline": false,
        "w": 6,
        "x": 0,
        "y": 2,
      },
      Object {
        "h": 8,
        "i": "19c7a585-82ac-4e83-bbd4-b8757b445c4e",
        "savedQuery": Array [
          Object {
            "label": "Sales",
            "value": "sales",
          },
        ],
        "sparkline": false,
        "w": 6,
        "x": 6,
        "y": 2,
      },
    ],
    "layout": Array [
      Object {
        "h": 1,
        "i": "19c7a585-82ac-4e83-bbd4-b8757b445c4e",
        "text": "<h1>Users and sales</h1>",
        "type": "paragraph",
        "w": 12,
        "x": 0,
        "y": 0,
      },
      Object {
        "h": 8,
        "i": "19c7a585-82ac-4e83-bbd4-b8757b445c4e",
        "savedQuery": Array [
          Object {
            "label": "Users",
            "value": "users",
          },
        ],
        "sparkline": false,
        "w": 6,
        "x": 0,
        "y": 2,
      },
      Object {
        "h": 8,
        "i": "19c7a585-82ac-4e83-bbd4-b8757b445c4e",
        "savedQuery": Array [
          Object {
            "label": "Sales",
            "value": "sales",
          },
        ],
        "sparkline": false,
        "w": 6,
        "x": 6,
        "y": 2,
      },
    ],
  },
}
`;
